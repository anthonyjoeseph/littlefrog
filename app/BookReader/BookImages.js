// app/BookReader/BookImages.js

import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  Text
} from 'react-native';

import {IndicatorViewPager, PagerDotIndicator} from 'rn-viewpager';

import BookImagesPage from './BookImagesPage'
class BookImages extends Component {
  render(){
    return (
      <IndicatorViewPager
                    style={
                      [
                        this.props.style,
                        {width: this.props.width, height: this.props.height}
                      ]
                    }
                    indicator={<PagerDotIndicator pageCount={this.props.pageFileNames.length} />}
                    onPageScroll={this.props.onPageScroll}>
          {
            this.props.pageFileNames.map(
              function(pageFileName, index){
                return (
                  <View key={pageFileName} style={{width:this.props.width, height:this.props.height}}>
                    <BookImagesPage
                      source={{uri: pageFileName}}
                      onQuizSelect={() => {this.props.onQuizSelect(index)}}
                      width={this.props.width}
                      height={this.props.height} />
                  </View>
                );
              }.bind(this)
            )
          }
      </IndicatorViewPager>
    );
  }
}

export default BookImages;
