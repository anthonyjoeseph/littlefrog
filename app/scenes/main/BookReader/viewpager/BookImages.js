// @flow
// app/BookReader/BookImages.js

import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  Text,
  Image,
  ScrollView
} from 'react-native';

import ViewPager from './ViewPager';
import BookImagesPage from './BookImagesPage';

function isNearInt(value, err) {
  var nearbyInt = Math.round(value);
  return Math.abs(value - nearbyInt) < 0.05;
}

class BookImages extends Component {

  render(){
    return (
      <View style={{width:this.props.width, height:this.props.height}}>
        <ViewPager
          numPages={this.props.pagesData.length}
          renderRange={4}
          onPageScroll={this.props.onPageScroll}>
            {
              this.props.pagesData.map(
                function(data, index){
                  return (
                    <BookImagesPage
                     key={index}
                     source={{
                       uri: this.props.baseURI + "pages/" + index + ".jpg",
                       headers: this.props.authHeaders
                     }}
                     animations={data.animations}
                     questionIcons={data.questionIcons}
                     onQuizSelect={
                       function(questionIndex){
                         this.props.onQuizSelect(index, questionIndex)
                       }.bind(this)
                     }
                     width={this.props.width}
                     height={this.props.height} />
                  );
                }.bind(this)
              )
            }
        </ViewPager>
      </View>
    );
  }
}
export default BookImages;
