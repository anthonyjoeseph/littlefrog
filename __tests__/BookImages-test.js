// __tests__/BookImages-test.js
import 'react-native';
import React from 'react';
import BookImages from '../app/scenes/main/BookReader/viewpager/BookImages';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  var testData = require('./testData/TestBookData.json');

  const tree = renderer.create(
    <BookImages
      style={{color:'red'}}
      onPageScroll={ () => {} }
      pagesData={ testData.pages }
      onQuizSelect={ () => {} }
      width={ 400 }
      height={ 200 } />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
