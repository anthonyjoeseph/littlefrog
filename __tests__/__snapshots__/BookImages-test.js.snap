exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "height": 200,
      "width": 400,
    }
  }>
  <ScrollView
    horizontal={true}
    onScroll={[Function]}
    pagingEnabled={true}
    scrollEventThrottle={1}>
    <View
      style={
        Object {
          "alignItems": "stretch",
          "flexDirection": "row",
          "height": 200,
          "justifyContent": "flex-start",
          "width": 400,
        }
      }>
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": "page1",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "width": null,
          }
        }>
        <Image
          source={
            Object {
              "uri": "spongebob.gif",
            }
          }
          style={
            Object {
              "bottom": 120,
              "height": 60,
              "left": 120,
              "position": "absolute",
              "width": 70,
            }
          } />
        <Image
          source={
            Object {
              "uri": "bored_animation.gif",
            }
          }
          style={
            Object {
              "bottom": 220,
              "height": 60,
              "left": 320,
              "position": "absolute",
              "width": 170,
            }
          } />
      </Image>
    </View>
    <View
      style={
        Object {
          "alignItems": "stretch",
          "flexDirection": "row",
          "height": 200,
          "justifyContent": "flex-start",
          "width": 400,
        }
      }>
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": "page2",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "width": null,
          }
        } />
    </View>
    <View
      style={
        Object {
          "alignItems": "stretch",
          "flexDirection": "row",
          "height": 200,
          "justifyContent": "flex-start",
          "width": 400,
        }
      }>
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": "page3",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "width": null,
          }
        }>
        <View
          style={
            Object {
              "bottom": 100,
              "height": 70,
              "left": 200,
              "position": "absolute",
              "width": 60,
            }
          }>
          <Image
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            resizeMode="stretch"
            source={
              Object {
                "uri": "frog_outline",
              }
            }
            style={
              Object {
                "height": 70,
                "tintColor": null,
                "width": 60,
              }
            }
            testID={undefined} />
        </View>
        <View
          style={
            Object {
              "bottom": 120,
              "height": 20,
              "left": 240,
              "position": "absolute",
              "width": 80,
            }
          }>
          <Image
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            resizeMode="stretch"
            source={
              Object {
                "uri": "frog_outline",
              }
            }
            style={
              Object {
                "height": 20,
                "tintColor": null,
                "width": 80,
              }
            }
            testID={undefined} />
        </View>
      </Image>
    </View>
    <View
      style={
        Object {
          "alignItems": "stretch",
          "flexDirection": "row",
          "height": 200,
          "justifyContent": "flex-start",
          "width": 400,
        }
      }>
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": "page4",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "width": null,
          }
        } />
    </View>
    <View
      style={
        Object {
          "alignItems": "stretch",
          "flexDirection": "row",
          "height": 200,
          "justifyContent": "flex-start",
          "width": 400,
        }
      }>
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": "page5",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "width": null,
          }
        } />
    </View>
    <View
      height={200}
      width={400} />
    <View
      height={200}
      width={400} />
    <View
      height={200}
      width={400} />
    <View
      height={200}
      width={400} />
    <View
      height={200}
      width={400} />
    <View
      height={200}
      width={400} />
    <View
      height={200}
      width={400} />
    <View
      height={200}
      width={400} />
    <View
      height={200}
      width={400} />
    <View
      height={200}
      width={400} />
  </ScrollView>
</View>
`;
